### PR TITLE
[sync2] Tiny test extension for more consistency

### PR DESCRIPTION
**Pull request by th4t**

### CODE REVIEW

File: sync2/semaphore_test.go [hunk 1]
```diff
@@ -133,6 +133,12 @@ func (suite *SemaphoreSuite) TestLotsOfWaiters(t *C) {
 	for found := 0; found < 1000; found++ {
 		<-c
```
Comment (PatrickDropbox) [inline sync2/semaphore_test.go]: could you also change the channel size to 1000 and change this to

```
select {
case <- c:
default:
    t.FailNow()
}
```

Comment (th4t) [inline sync2/semaphore_test.go]: That's funny, the second change makes the test fail. Inserting a tiny sleep helps over this induced racing condition. I am confused. Increment appears to be blocking.
```
    s.Increment(2000)
    time.Sleep(time.Duration(1)*time.Nanosecond)
```

**Comment by PatrickDropbox** [comment]
sleep make sense.  the Increment triggers a condition bcast, but does not yield context to other goroutines

**Comment by th4t** [comment]
Cool, thanks for the explanation and quick merge!

**Pull request merged**